mypyc/codegen/emit.py

```diff
@@ -695,6 +695,11 @@ def emit_box(self, src: str, dest: str, typ: RType, declare_dest: bool = False,
             self.emit_lines('{}{} = Py_None;'.format(declaration, dest))
             if not can_borrow:
                 self.emit_inc_ref(dest, object_rprimitive)
+        # TODO: This is a hack to handle mypy's false negative on unreachable code.
+        #       All ops returning int32/int64 should not be coerced into object.
+        #       Since their result will not be used elsewhere, it's safe to use NULL here
+        elif is_int32_rprimitive(typ) or is_int64_rprimitive(typ):
+            self.emit_lines('{}{} = NULL;'.format(declaration, dest))
```

Comment:
here's the hack

Reply:
This is fine for now, but can you create a follow-up issue to properly box these as Python `int` objects? It should be easy enough to support, and it could be useful in the future.